Specifications/Applications/for_Application/when_comparing_with_different_structures.cs

```diff
@@ -0,0 +1,31 @@
+using Machine.Specifications;
+
+namespace Dolittle.Applications.Specs.for_Application
+{
+    public class when_comparing_with_different_structures
+    {
+        const string name = "Application";
+
+        static IApplication application_a, application_b;
+
+        Establish context = () =>
+        {
+            application_a = Application.WithName(name)
+                .WithStructureStartingWith<BoundedContext>(b => b.Required
+                    .WithChild<Module>(m => m.Required
+                        .WithChild<Feature>()))
+                .Build();
+                
+            application_b = Application.WithName(name)
+                .WithStructureStartingWith<BoundedContext>(b => b.Required
+                    .WithChild<Module>(m => m.Required))
+                .Build();
+
+            
+        };
+        It should_not_be_the_same = () => application_a.Equals(application_b).ShouldBeFalse();
```

Comment:
You are testing equality / gethashcode, not comparison.

I also prefer to use the Because of = () => { action } to do the actual action under test rather than just fold it into the assert.

so you should have a variable called is_equal and assigned to that in the Because and then assert against it.  Normally (and I probably would) I'd have two separate specs for equality and GetHashCode but I can see the argument for doing both actions as the setup has to be exactly the same.